container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/dump_kurtosis.go

```diff
@@ -0,0 +1,69 @@
+package engine_functions
+
+import (
+	"context"
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/stacktrace"
+	"os"
+	"path"
+	"strings"
+)
+
+const (
+	engineLogsSubDirpathSuffix = "engines"
+	createdDirPerms            = 0755
+	enclavesSubDirpathFragment = "enclaves"
+	enclaveNameUuidSeparator   = "--"
+	errorSeparator             = "\n\n"
+)
+
+var allEnclavesFilter = &enclave.EnclaveFilters{UUIDs: nil, Statuses: nil}
+
+func DumpKurtosis(ctx context.Context, outputDirpath string, backend backend_interface.KurtosisBackend) error {
+
+	allEnclaves, err := backend.GetEnclaves(ctx, allEnclavesFilter)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while getting a list of enclaves registered with the underlying engine")
+	}
+
+	// Create the main output directory
+	if _, err = os.Stat(outputDirpath); !os.IsNotExist(err) {
```

Comment:
Looking at the code, `os.IsNotExist` I think will panic when err is nil here. But also surprised you didn't catch that in your tests, so I might be reading this wrong. Anyways, can you maybe just do `err != nil && !os.IsNotExist(err)` to avoid any confusion ?